File: 4_spring-boot-data-jpa/src/main/java/com/example/springdatajpa/demo/entity/Delivery.java
package com.example.springdatajpa.demo.entity;

import com.example.springdatajpa.demo.entity.status.DeliveryStatus;
import jakarta.persistence.*;
import lombok.Getter;

@Entity
@Getter
public class Delivery {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "delivery_id")
    private Long id;

    @OneToOne(mappedBy = "delivery")
    private Order order;

    @Embedded
    private Address address;

    private DeliveryStatus status;
}

File: spring-boot-jpa2-restapi/restapi/src/main/kotlin/com/example/restapi/domain/Delivery.kt
package com.example.restapi.domain

import jakarta.persistence.*

@Entity
class Delivery(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "delivery_id")
    var id: Long? = null,

    @OneToOne(mappedBy = "delivery", fetch = FetchType.LAZY)
    var order: Order? = null,

    @Embedded
    var address: Address,

    @Enumerated(EnumType.STRING)
    var status: DeliveryStatus
)